src/main/java/frc/robot/poseestimator/helpers/ObservationCountHelper.java

```diff
@@ -0,0 +1,38 @@
+package frc.robot.poseestimator.helpers;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Supplier;
+
+public class ObservationCountHelper<T> {
+
+	private final Supplier<List<T>> observationSupplier;
+	private final int maxCount;
```

Comment:
![image](https://github.com/user-attachments/assets/49d84088-76a6-474c-a250-85a7465cc357)
(jk, I'm fine with max in the name of the var)
But, be more precise; the maximum of what? By this naming, it could be the max size of the stack, or the max time of savings, etc... (ans it's none of those).